app/services/imports/lettings_logs_field_import_service.rb

```diff
@@ -203,5 +205,103 @@ def update_reason(xml_doc)
         @logger.warn("lettings log with old id #{old_id} not found")
       end
     end
+
+    def update_person_details(xml_doc)
+      return if meta_field_value(xml_doc, "form-name").include?("Sales")
+
+      old_id = meta_field_value(xml_doc, "document-id")
+      record = LettingsLog.find_by(old_id:)
+
+      return @logger.warn("lettings log with old id #{old_id} not found") unless record
+      return @logger.info("lettings log #{record.id} has no hhmemb, skipping update") unless record.hhmemb
+
+      if (2..record.hhmemb).all? { |person_index| record.has_any_person_details?(person_index) || record["details_known_#{person_index}"] == 1 }
+        return @logger.info("lettings log #{record.id} has all household member details, skipping update")
+      end
+      if record.hhmemb == 8 || (record.hhmemb + 1..8).none? { |person_index| file_contains_person_details?(xml_doc, person_index) }
```

Comment:
can you put record.hhemb + 1 in brackets so it's clearer it's a lower bound of this array or is that not linty? I think it's quite tricky to read because it at first seems like we're adding record.hhemb to 1..8 but if this is standard notation and I've just not come across it then fine :D